src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import core.basesyntax.models.*;
+import java.util.Random;
+
+class FigureSupplier {
+    private final ColorSupplier colorSupplier;
+    private final Random random;
+
+    public FigureSupplier() {
+        this.colorSupplier = new ColorSupplier();
+        this.random = new Random();
+    }
+
+    public Figure getRandomFigure() {
+        int type = random.nextInt(5);

```

Comment:
All magic numbers in your code should be constants. The number 5 should be extracted into a constant with a clear name, for example, FIGURE_COUNT.